RecoHGCal/TICL/plugins/SeedingRegionByTracks.cc

```diff
@@ -0,0 +1,74 @@
+// Author: Arabella Martelli, Felice Pantaleo, Marco Rovere - arabella.martelli@cern.ch, felice.pantaleo@cern.ch, marco.rovere@cern.ch
+// Date: 06/2019
+#include <algorithm>
+#include <set>
+#include <vector>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "SeedingRegionByTracks.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateTransform.h"
+
+using namespace ticl;
+
+SeedingRegionByTracks::SeedingRegionByTracks(const edm::ParameterSet &conf, edm::ConsumesCollector &sumes)
+    : SeedingRegionAlgoBase(conf, sumes),
+      cutTk_(conf.getParameter<std::string>("cutTk")),
+      propName_(conf.getParameter<std::string>("propagator")) {
+  tracks_token_ = sumes.consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracks"));
+}
+
+SeedingRegionByTracks::~SeedingRegionByTracks(){};
+
+void SeedingRegionByTracks::makeRegions(const edm::Event &ev,
+                                        const edm::EventSetup &es,
+                                        std::vector<ticl::TICLSeedingRegion> &result) {
+  std::call_once(initializeGeometry_, [&]() {
```

Comment:
this doesn't look right: other instances of SeedingRegionByTracks in other threads should have bfield_ etc initialized;  IOVs can change.
Use ESWatchers instead.

Reply:
replaced with beginRun